tests/unit/src/test/scala/tests/worksheets/WorksheetSuite.scala

```diff
@@ -238,4 +247,71 @@ class WorksheetSuite extends FunSuite {
        |""".stripMargin
   )
 
+  val testTypesString =
+    """
+      |val a = 1
+      |val b: String = "A"
+      |val (x, y) = (true, 0)
+      |a + 4
+      |class C {}
+      |def f(): String = "FFF"
+      |f()
+      |
+      |def px(): String = { println("Something"); "OK" }
+      |
+      |println("print")
+      |
+      |px()
+      |
+      |{ println("P1"); List(1,2,3) }
+      |""".stripMargin
+
+  checkDecorations(
+    "types-enabled",
+    testTypesString,
+    """|
+       |<val a = 1> // 1: Int
+       |<val b: String = "A"> // "A": String
+       |<val (x, y) = (true, 0)> // x: Boolean=true, y: Int=0
+       |<a + 4> // 5: Int
+       |class C {}
+       |def f(): String = "FFF"
+       |<f()> // "FFF": String
+       |
+       |def px(): String = { println("Something"); "OK" }
+       |
+       |<println("print")> // _: Unit , out: print
```

Comment:
I wonder if it's better to leave the type out o `Unit` binders?

Reply:
I don't like inconsistent behaviour but in this case it probably makes better sense.
Reverted to previous state where only output is printed.